EventListener/Login.php

```diff
@@ -63,6 +67,28 @@ public function setCustomerId($value)
         $this->options['customerId'] = $value;
     }
 
+    /**
+     * Return an ID or the Username if no ID available.
+     *
+     * @param TokenInterface $token
+     *
+     * @return mixed
+     */
+    protected function getUserIdentifier(TokenInterface $token)
+    {
+        $tokenUser = $token->getUser();
+
+        if ($tokenUser instanceof eZUser || !$tokenUser instanceof UserInterface) {
+            return $tokenUser->getAPIUser()->id;
+        }
+
+        if (method_exists($tokenUser, 'getId')) {
+            return $tokenUser->getId();
+        }
+
+        return $tokenUser->getUsername();
```

Comment:
Thank you for details @johaho.

@Plopix can we ask you to modify also this part https://github.com/ezsystems/EzSystemsRecommendationBundle/blob/master/Twig/RecommendationTwigExtension.php#L310  which is responsible for injecting userId into tracking code? user ID should be the same in both places.